proto/identity/associations/association.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package xmtp.identity.associations;
+
+import "identity/associations/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/identity/associations";
+option java_package = "org.xmtp.proto.identity.associations";
+
+// The first entry of any XID log, which is considered valid iff the XID can be
+// deterministically derived from the address and nonce.
+// The recovery address defaults to the initial_associated_address unless
+// there is a subsequent ChangeRecoveryAddress in the log.
+message CreateXid {
+  string xid = 1;
+  uint32 nonce = 2;
+  string initial_associated_address = 3;
+}
+
+message AddAssociation {
+  enum AssociatedRole {
+    ASSOCIATED_ROLE_UNSPECIFIED = 0;
+    ASSOCIATED_ROLE_INSTALLATION = 1;
+    ASSOCIATED_ROLE_ADDRESS = 2;
+    ASSOCIATED_ROLE_LEGACY_KEY = 3;
+  }
+
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  uint32 nonce = 3;
+  Signature existing_member_signature = 4;
+  Signature new_member_signature = 5;
+  AssociatedRole new_member_role = 6;
+}
+
+// An entity may have been associated with multiple roles. Clients MUST remove
+// the entity from all previously associated roles.
+message RevokeAssociation {
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  uint32 nonce = 3;
+  Signature recovery_address_signature = 4;
+  // The nonce of the association being removed.
+  // This nonce must never be used again for this member.
+  uint32 revoked_nonce = 4;
```

Comment:
Could this be a hash of the whole association?

Reply:
I guess if we used timestamps instead of nonces/hashes, for this field and the `retained_associations` field, the signature text would be more comprehensible to the user. Not sure how much this matters

```
Revoke 0xabcdef
Which was created at: <timestamp>
The following associations will remain valid:
0x234324 created at: <timestamp>
0x656734 created at: <timestamp>
```

Reply:
I do like the human readability of it being a timestamp.